ayon_server/entities/folder.py

```diff
@@ -67,160 +68,146 @@ async def load(
+                        updated_at=datetime.now(),
+                    )
                 )
-            )
 
-        else:
-            # Create a new entity
-            await transaction.execute(
-                *SQLTool.insert(
-                    f"project_{self.project_name}.{self.entity_type}s",
-                    **dict_exclude(self.dict(exclude_none=True), ["own_attrib"]),
+            else:
+                # Create a new entity
+                await Postgres.execute(
+                    *SQLTool.insert(
+                        f"project_{self.project_name}.{self.entity_type}s",
+                        **dict_exclude(self.dict(exclude_none=True), ["own_attrib"]),
+                    )
                 )
-            )
 
-    async def commit(self, transaction: Connection | None = None) -> None:
+            if auto_commit:
+                await self.commit()
+
+    async def commit(self) -> None:
         """Refresh hierarchy materialized view on folder save."""
 
-        async def _commit(conn):
-            await conn.execute(
+        async with Postgres.transaction():
+            await Postgres.execute(
                 f"""
                 REFRESH MATERIALIZED VIEW CONCURRENTLY
                 project_{self.project_name}.hierarchy
                 """
             )
-            await rebuild_inherited_attributes(self.project_name, transaction=conn)
+            await rebuild_inherited_attributes(self.project_name)
             await rebuild_hierarchy_cache(self.project_name)
 
-        if transaction is not None:
-            await _commit(transaction)
-            return
-        else:
-            async with Postgres.acquire() as conn, conn.transaction():
-                await _commit(conn)
-
-    async def delete(self, transaction: Connection | None = None, **kwargs) -> bool:
-        if not transaction:
-            async with Postgres.acquire() as conn, conn.transaction():
-                return await self._delete(conn, **kwargs)
-        else:
-            return await self._delete(transaction, **kwargs)
-
-    async def _delete(self, transaction: Connection, **kwargs) -> bool:
-        if kwargs.get("force", False):
-            logger.info(f"Force deleting folder and all its children. {self.path}")
-            await transaction.execute(
-                f"""
-                DELETE FROM project_{self.project_name}.products
-                WHERE folder_id IN (
-                    SELECT id FROM project_{self.project_name}.hierarchy
-                    WHERE path = $1
-                    OR path LIKE $1 || '/%'
-                ) RETURNING name
-                """,
-                self.path.lstrip("/"),
-            )
+    async def delete(self, *args, auto_commit: bool = True, **kwargs) -> bool:
+        async with Postgres.transaction():
+            if kwargs.get("force", False):
+                logger.info(f"Force deleting folder and all its children. {self.path}")
```

Comment:
I assume `:` may make more sense on this logged message?
```suggestion
                logger.info(f"Force deleting folder and all its children: {self.path}")
```